js/packages/mobile-wallet-adapter-protocol/src/types.ts

```diff
@@ -85,7 +85,7 @@ export interface AuthorizeAPI {
         features?: IdentifierArray; 
         addresses?: string[]; 
         auth_token?: AuthToken; 
-        sign_in_payload?: SolanaSignInInput;
+        sign_in_payload?: Pick<SignInPayload, 'domain' | 'uri'> & Partial<SignInPayload>;
```

Comment:
IIUC, the `SignInPayload` type is used 'internally' by the SDK, but this utility type will be used by exposed/consumed by devs.

For type readability and usability by devs, I'd suggest moving this into a public type like:

```
export type SolanaSignInPayload = Pick<SignInPayload, 'domain' | 'uri'> & Partial<SignInPayload>;
```

Similar to how `Chain`, `AuthToken`, etc are defined. Type name is more readable and more importantly, devs can import and use the type.

Reply:
SignInPayload can/will be used publicy as well. 

but yeah I can move this to a type like [`SolanaSignInInputWithRequiredFields`](https://github.com/solana-labs/wallet-standard/blob/2a1d55c9a57170f8ccecbb2f81f070442e7bd670/packages/core/util/src/signIn.ts#L56C48-L56C48)